src/Jobs/JobsAll.tsx

```diff
@@ -0,0 +1,105 @@
+// React components
+import React, { useCallback, useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+// Material UI components
+import { useTheme } from '@mui/material/styles';
+import { TableCell } from '@mui/material';
+
+// Local data
+import JobsBase, { Job, headerStyles } from './JobsBase';
+
+const JobsAll: React.FC = () => {
+  const fiaApiUrl = process.env.REACT_APP_FIA_REST_API_URL;
+  const theme = useTheme();
+
+  const [jobs, setJobs] = useState<Job[]>([]);
+  const [totalRows, setTotalRows] = useState(0);
+  const [currentPage, setCurrentPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [orderDirection, setOrderDirection] = useState<'asc' | 'desc'>('desc');
+  const [orderBy, setOrderBy] = useState<string>('run_start');
+
+  const fetchTotalCount = useCallback(async (): Promise<void> => {
+    try {
+      const response = await fetch(`${fiaApiUrl}/jobs/count`);
+      const data = await response.json();
+      setTotalRows(data.count);
+    } catch (error) {
+      console.error('Error fetching run count:', error);
+    }
+  }, [fiaApiUrl]);
+
+  const fetchJobs = useCallback(async (): Promise<void> => {
+    try {
+      const token = localStorage.getItem('scigateway:token');
+      const offset = currentPage * rowsPerPage;
+      const query = `limit=${rowsPerPage}&offset=${offset}&order_direction=${orderDirection}&include_run=true`;
+      const response = await fetch(`${fiaApiUrl}/jobs?${query}`, {
+        method: 'GET',
+        headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${token}` },
+      });
+      const data = await response.json();
+
+      // Filter out any reductions that don't have a valid run object
+      const filteredData = data.filter((job: Job) => job.run !== null);
```

Comment:
I had added that line a few weeks ago. From memory, the page would crash when one of these runs would come up as it would try looking for an experiment ID that doesn't exist.

So should I add a check for these simple jobs and add them to the all table then?

Reply:
How will it know what type it is, will this be added to the API or will I need to add logic to the frontend to determine its type?

Reply:
Mmm yes, it's not in the API but is in the DB at this time, how would you like it to be returned by the API? as a string or some representation of the Enum?